src/rodeo_server.rs

```diff
@@ -37,8 +37,8 @@ impl RodeoServer {
   }
 
   /// Get the current GoatRodeoCluster from the the server
-  pub fn get_cluster(&self) -> GoatRodeoCluster {
-    let the_cluster: GoatRodeoCluster = (&(**self.get_cluster_arcswap().load())).clone();
+  pub fn get_cluster(&self) -> Arc<GoatRodeoCluster> {
+    let the_cluster: Arc<GoatRodeoCluster> = (&(*self.get_cluster_arcswap().load())).clone();
```

Comment:
This smells funny, I might unpack that later and see if I can figure out why the reference hackery is needed. Looks like it does the right thing though so shipit

Reply:
It's because we need to pass `Arc<GoatRodeoCluster>` around... and `clone()`ing an `Arc` is much cheaper than cloning the underlying data structure.

Reply:
okay but the reference and pointer creation isn't needed:
```suggestion
    let the_cluster: Arc<GoatRodeoCluster> = self.get_cluster_arcswap().load().clone();
```